controlplane/src/core/repositories/SubgraphRepository.ts

```diff
@@ -733,7 +734,7 @@ export class SubgraphRepository {
     return subgraphs;
   }
 
-  private async getSubgraph(conditions: SQL<unknown>[]): Promise<SubgraphDTO | undefined> {
+  private async getSubgraph(conditions: SQL<unknown>[]): Promise<SubgraphDTO | FeatureSubgraphDTO | undefined> {
```

Comment:
I'd suggest to split it into `getSubgraph` and `getFeatureSubgraph` repositories. This avoid fetching the feature subgraph for all calls and avoids that the logic becomes more complex.